plugins/@grouparoo/pipedrive/src/lib/common/destinationMappingOptions.ts

```diff
@@ -90,80 +107,73 @@ const mapTypesToGrouparoo = (
   return null;
 };
 
-type KnownPersonField = {
-  key: string;
-  pipedriveKey: string;
-  type: DestinationMappingOptionsResponseType;
-  required: boolean;
-  important: boolean;
-  groupMembershipField: boolean;
-};
-
-export const getKnownPersonFieldMap = async (
+export const getKnownFields = async (
   client: PipedriveClient,
+  type: EntityType,
   cacheData: PipedriveCacheData,
   update: boolean = false
-): Promise<{ [fieldName: string]: string }> => {
-  const known = await getKnownPersonFields(client, cacheData, update);
-  const out = {};
-  for (const field of known) {
-    out[field.key] = field.pipedriveKey;
-  }
-  return out;
-};
-
-const getKnownPersonFields = async (
-  client: PipedriveClient,
-  cacheData: PipedriveCacheData,
-  update: boolean = false
-): Promise<KnownPersonField[]> => {
+): Promise<KnownPipedriveField[]> => {
   const { appId, appOptions } = cacheData;
   const cacheDurationMs = 1000 * 60 * 10; // 10 minutes
-  const cacheKey = ["getKnownPersonFields", appOptions];
+  const cacheKey = [configMapping[type].cacheKey, appOptions];
```

Comment:
you could simplify here without the special `cacheKey` and do
`["getKnownFields", type, appOptions];`